### PR TITLE
Prevent the TE fuser from getting datatypes it can't handle

### PR DESCRIPTION
**Pull request by bertmaher**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44152 Fix frac in CUDA fuser
* **#44160 Prevent the TE fuser from getting datatypes it can't handle**
* #44157 [te] Fix casting of unsigned char, and abs(int)

Differential Revision: [D23528508](https://our.internmc.facebook.com/intern/diff/D23528508)

### CODE REVIEW

File: test/test_jit_fuser_te.py [hunk 1]
```diff
@@ -1210,7 +1213,8 @@ def rand(dtype, device="cuda"):
                 continue
             t = torch.jit.trace(fn, (x,))
             self.assertEqual(ref, t(x))
-            self.assertAllFused(t.graph_for(x))
+            if dtype in supported_dtypes:
```
Comment (ZolotukhinM) [inline test/test_jit_fuser_te.py:1216]: What do we test when we run scripted but not fused graphs (when the dtype is unsupported)?

File: test/test_jit_fuser_te.py [hunk 2]
```diff
@@ -1146,24 +1146,27 @@ def rand(dtype, device="cuda"):
...
 
-        dtypes = [
+        supported_dtypes = [
             torch.int8,
             torch.uint8,
             torch.int16,
             torch.int32,
             torch.int64,
             # torch.float16,
-            # torch.bfloat16,
```
Comment (ZolotukhinM) [inline test/test_jit_fuser_te.py:1156]: Is it intentional that `torch.float16` remains commented out?

**Review by ZolotukhinM — APPROVED** [review]
👍

File: test/test_jit_fuser_te.py (hunk 1 above)
Comment (bertmaher) [reply test/test_jit_fuser_te.py:1216]: Basically this is just testing that we don't generate something that throws (which is what would happen before this diff).  But it's kind of ridiculous to test that for every single unary op.  Let me split the unsupported dtypes into a separate tiny test.

File: test/test_jit_fuser_te.py (hunk 2 above)
Comment (bertmaher) [reply test/test_jit_fuser_te.py:1156]: Yeah, some fixes for float16 are in the works.  This should probably be a TODO but I expect it to land in a day or so.

**Review by agniutkarsh — APPROVED** [review]

**Pull request closed without merging**